src/execution_plan/ops/op_cond_var_len_traverse.c

```diff
@@ -102,7 +128,16 @@ Record CondVarLenTraverseConsume(OpBase *opBase) {
     Node n = Path_pop(p);
     Path_free(p);
 
-    Record_AddNode(op->r, op->destNodeIdx, n);
+    if(op->expandInto) {
+        /* Dest node is already resolved
+         * need to make sure src is connected to dest
+         * i.e. n == dest. */
+        Node *destNode = Record_GetNode(op->r, op->destNodeIdx);
+        if(ENTITY_GET_ID(&n) != ENTITY_GET_ID(destNode)) goto compute_path;
```

Comment:
I don't love `goto` tags for purposes other than cleanup, but this is an effective solution.